action.yml

```diff
@@ -0,0 +1,310 @@
+        
+        echo "Downloading wasixcc from: $DOWNLOAD_URL"
+        if [ -n "$GITHUB_TOKEN" ]; then
+          curl -L -H "Authorization: token $GITHUB_TOKEN" -o wasixcc.tar.gz "$DOWNLOAD_URL"
+        else
+          curl -L -o wasixcc.tar.gz "$DOWNLOAD_URL"
+        fi
+        
+        if [ ! -f wasixcc.tar.gz ]; then
+          echo "Failed to download wasixcc"
+          exit 1
+        fi
+        
+        echo "Downloaded wasixcc successfully"
+
+    - name: Install wasixcc
+      id: install-wasixcc
+      shell: sh
+      run: |
+        echo "Extracting wasixcc..."
+        tar -xzf wasixcc.tar.gz
+        rm wasixcc.tar.gz
+        
+        # Create installation directory for the main wasixcc binary
+        WASIXCC_HOME="$HOME/.wasixcc"
+        mkdir -p "$WASIXCC_HOME"
+        
+        # Create bin directory for executables installed by --install-executables
+        INSTALL_DIR="$WASIXCC_HOME/bin"
+        mkdir -p "$INSTALL_DIR"
+        
+        # Move binary to .wasixcc (not bin) to avoid --install-executables overwriting it
+        if [ "${{ steps.platform.outputs.os }}" = "windows" ]; then
+          mv wasixcc.exe "$WASIXCC_HOME/"
+          WASIXCC_PATH="$WASIXCC_HOME/wasixcc.exe"
+        else
+          mv wasixcc "$WASIXCC_HOME/"
+          chmod +x "$WASIXCC_HOME/wasixcc"
+          WASIXCC_PATH="$WASIXCC_HOME/wasixcc"
+        fi
+        
+        # Add bin directory to PATH
+        printf "%s\n" "$INSTALL_DIR" >> $GITHUB_PATH
+        
+        # Install executables (wasix++, wasixar, etc.) to bin directory
+        echo "Installing wasixcc executables..."
+        "$WASIXCC_PATH" --install-executables "$INSTALL_DIR"
+        
+        printf "path=%s\n" "$WASIXCC_PATH" >> $GITHUB_OUTPUT
+        echo "Installed wasixcc to: $WASIXCC_PATH"
+        
+        # Verify installation
+        "$WASIXCC_PATH" --version
+
+    - name: Install binaryen
+      if: inputs.install-binaryen == 'true'
+      shell: sh
+      env:
+        GITHUB_TOKEN: ${{ inputs.github_token }}
+      run: |
+        echo "Installing binaryen (wasm-opt)..."
+        BINARYEN_TAG="${{ inputs.binaryen-tag }}"
+        
+        if [ "$BINARYEN_TAG" = "latest" ]; then
+          "$HOME/.wasixcc/wasixcc" --download-binaryen
+        else
+          "$HOME/.wasixcc/wasixcc" --download-binaryen "$BINARYEN_TAG"
+        fi
+        
+        echo "Installed binaryen successfully"
+    
+    - name: Add binaryen to PATH
+      if: inputs.install-binaryen == 'true' && inputs.add-binaryen-path == 'true'
+      shell: sh
+      run: |
+        # Find binaryen installation directory
+        BINARYEN_DIR=$(find "$HOME/.wasixcc" -type d -name "binaryen-*" | head -n 1)
+        if [ -n "$BINARYEN_DIR" ] && [ -d "$BINARYEN_DIR/bin" ]; then
+          printf "%s\n" "$BINARYEN_DIR/bin" >> $GITHUB_PATH
+          echo "Added binaryen to PATH: $BINARYEN_DIR/bin"
+        else
+          echo "Warning: Could not find binaryen installation directory"
```

Comment:
If install-binaryen was set and pathadding was requested and we cant find it it is an exit 1